From 857ac4b0682dec32ba55e4080fa7ec560a2481a0 Mon Sep 17 00:00:00 2001
From: "Owen W. Taylor" <otaylor@fishsoup.net>
Date: Fri, 20 Nov 2009 10:19:03 -0500
Subject: [PATCH] Accept an empty string as well as "disabled" for
 keybindings

Treat the empty string the same as "disabled" for GConf keybinding
keys. gnome-keybinding-properties was changed to write disabled
keys as the empty string a year or so ago.

https://bugzilla.gnome.org/show_bug.cgi?id=559816
---
 src/ui/ui.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ui/ui.c b/src/ui/ui.c
index 78aa353..ef9fcfc 100644
--- a/src/ui/ui.c
+++ b/src/ui/ui.c
@@ -954,7 +954,7 @@ meta_ui_parse_accelerator (const char          *accel,
   *keycode = 0;
   *mask = 0;
 
-  if (strcmp (accel, "disabled") == 0)
+  if (!accel[0] || strcmp (accel, "disabled") == 0)
     return TRUE;
   
   meta_ui_accelerator_parse (accel, &gdk_sym, &gdk_code, &gdk_mask);
@@ -1041,7 +1041,7 @@ meta_ui_parse_modifier (const char          *accel,
   
   *mask = 0;
 
-  if (accel == NULL || strcmp (accel, "disabled") == 0)
+  if (accel == NULL || !accel[0] || strcmp (accel, "disabled") == 0)
     return TRUE;
   
   meta_ui_accelerator_parse (accel, &gdk_sym, &gdk_code, &gdk_mask);
-- 
1.7.9

